Add 2022-08-18 agenda and minutes (#68)
solid/notifications-panel: meetings/2022-08-18.md

* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/)
* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Ethics and Professional Conduct](https://www.w3.org/Consortium/cepc/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! please introduce yourself.

### Scribes
* Aaron
* Sarven

### Introductions

* name: text

---

## Topics

### WIP Implementation Feedback
* SC: We'll allocate some time for implementation feedback going forward. Links to servers/applications welcome.

### Subscription Request Response
URL: https://github.com/solid/notifications/pull/72

* SC: Anything outstanding? Let's wrap this up if nothing major.
* SC: This was clarified via https://github.com/solid/notifications/commit/29ad87f865f734f765b52d2f799cef3dfe50e00e

> * SC: For minimal interop, JSON-LD MUST be supported, but if an Accept header is sent, and if the server supports the requested serialization, that is what will be produced by the server.

Reviewer: ```suggestion
* SC: For minimal interop, JSON-LD MUST be supported, but if an `Accept` header is sent, and if the server supports the requested serialization, that is what will be produced by the server.
```

* SC: For JSON-LD payloads, this minimal serialization requires `@context` and `type`.
* SC: What appears to be missing is whether, for other serializations, the `type` and `topic` predicates need to be included. Perhaps this should be added to the description of the payload. The rest can be specific to JSON-LD
* AC: +1
* SC: can we merge this PR and then make that change in a separate PR?
* AC: Yes, that makes sense, I have already approved this PR

ACTION: SC to merge this PR and will make a follow-on PR to address the required triples for non-JSON-LD serializations.

### Interchangeable use of notification channel type and subscription type
URL: https://github.com/solid/notifications/issues/71

>* SC: I think we can/should review the modeling around this.
>* SC: I think this is a bug. Lets come back to it with responses...

* SC: Thoughts?

> * AC: If we use rdf:type in JSON-LD, it says that it is a subscription channel. That is a subscription channel and also a `WebsocketSubscription2021`, then there can be features that can come with it. Scoping the contexts. Then we can have websocket foo, bar, and those would be part of the subscription mechanism. channel is generic but to particular type. we could call it subscription channel along the lines of christoph's suggestion. if it is even about a subscription, we call it that.. and then we use a different property which would be the endpoint where you define the subscription. it seems to be a matter of clarifying the meaning of these properties and how they express what they're trying to express.

Reviewer: ```suggestion
* AC: If we use `rdf:type` in JSON-LD, it says that it is a subscription channel. That is a subscription channel and also a `WebsocketSubscription2021`; then there can be features that can come with it. Scoping the contexts. Then we can have websocket foo, bar, and those would be part of the subscription mechanism. Channel is generic but no particular type. We could call it subscription channel along the lines of christoph's suggestion. If it is even about a subscription, we call it that... and then we use a different property which would be the endpoint where you define the subscription. It seems to be a matter of clarifying the meaning of these properties and how they express what they're trying to express.
```

* SC: We may need to revisit the data modeling.

> ACTION: Find consensus on the solution to this, producing a proposal and make a PR by next meeting (2022-08-25)

Reviewer: ```suggestion
ACTION: Find consensus on the solution to this, produce a proposal, and make a PR by next meeting (2022-08-25)
```

### Solid Notifications Protocol: Review notes
URL: https://github.com/solid/notifications/issues/76

* SC: Talked with Tim about it (some specifics) but I didn't process the notes/review yet.
* SC: Can we all commit to addressing this review towards the next release of the Notifications Protocol (and WebSocket Subscription Type)?
* SC: Tim would like a simpler protocol. The thing that Tim seems to want does not seem realistic for the next release (e.g. GET on resource and the response header would contain specific WS-related connection information)

ACTION: SC and AC will process and respond to this issue (some items may already have been addressed)

### An Alternative (WebSockets) Design
URL: https://github.com/solid/notifications/issues/97

* RG: RFC. How are we dealing with multiple-resource connections?

> * AC: Lots of ways to address that. Part of the subscription negotiation you can assert a request to listen to arbitrary number of resources. If you then as part of the subscription negotiation, say I want to subscribe to 500 or whatever resources, you can negotiate an access token, to have at least read access to those.. there are different ways to check for that too - different protocols in place. You can even write an implementation that uses id tokens from Solid OIDC. Couple your authz system and websocket system. You can absolutely have 1 connection and listen to multiple resources at the same time.

Reviewer: ```suggestion
* AC: Lots of ways to address that. As part of the subscription negotiation, you can assert a request to listen to an arbitrary number of resources. If you then say, as part of the subscription negotiation, "I want to subscribe to 500 or whatever resources," you can negotiate an access token, to have at least READ access to those. There are different ways to check for that too — different protocols in place. You can even write an implementation that uses id tokens from Solid OIDC. Couple your authz system and websocket system. You can absolutely have 1 connection and listen to multiple resources at the same time.
```

> * SC: Remind me.. Is that by way of having an array for `topic`?

Reviewer: ```suggestion
* SC: Remind me. Is that by way of having an array for `topic`?
```

* AC: Indeed. Starting point. And then. Just need a way of negotiation is supported by the authorization server or the websocket server itself.
* SC: ...

> * RG: One or two alternatives are possible. ??? or doing it in band in the connection itself. suppose a resource gets added, do I then have to create a separate connection or is it in band? How will it be managed for addition and removal of resources. At least we have one method. That is clearly stated.

Reviewer: ```suggestion
* RG: One or two alternatives are possible. ??? or doing it in band in the connection itself. Suppose a resource gets added; do I then have to create a separate connection, or is it in band? How will it be managed for addition and removal of resources? At least we have one method that is clearly stated.
```

* AC: The way I'd do this is: first, if websockets makes use of bidirectional flow, or redo this negotiation inside the connection, then websockets is unlike any subscription mechanism. that's what's problematic about this. if however you do this on the subscription level and you want to upgrade, then you can reconnect. then whatever changed between disconnecting and connecting comes through. but for in-band it describes a separate protocol. so we should avoid that.
* RG: Only websockets is bidirectional.

> * AC: Just because websockets has a mechanism it doesn't mean it is supported in ??? the notifications protocol is trying to be generic. in order to be generic we can't specify / customise for each.

Reviewer: ```suggestion
* AC: Just because websockets has a mechanism doesn't mean it is supported in ???. The notifications protocol is trying to be generic. To be generic, we can't specify / customise for each.
```

> * RG: This point on generality we may be giving up on efficiency. If for 2.0 protocol, people can look at this. this remains in our memory.

Reviewer: ```suggestion
* RG: This point on generality, we may be giving up on efficiency. If for 2.0 protocol, people can look at this. This remains in our memory.
```

ACTION: RG to create issue to note the cases in which client may want to update its subscriptions or connections.

> * AC: For RG to consider, one thing that's important for protocol design is that you can layer. For the kind of thing you are describing re inside the websocket connection, but you could create a service that interacts and calls out a service... inside the service, the client can add/remove/manage whatever mechanism it wants. That service talks to Solid Notifications Protocol service.. and so from a client's perspective, they are doing everything inside the websocket. then you are free at that point because of working on the higher layer of the spec. you can get what you want without having the notifications protocol become more complicated. that's what happens with protocol.. and someone else comes with something that builds on that.

Reviewer: ```suggestion
* AC: For RG to consider, one thing that's important for protocol design is that you can layer. For the kind of thing you are describing re inside the websocket connection, but you could create a service that interacts and calls out a service... Inside the service, the client can add/remove/manage whatever mechanism it wants. That service talks to Solid Notifications Protocol service. From a client's perspective, they are doing everything inside the websocket. You are free at that point because of working on the higher layer of the spec. You can get what you want without having the notifications protocol become more complicated. That's what happens with protocol... and someone else comes with something that builds on that.
```

### Discovery of Notification Subscription Metadata
URL: https://github.com/solid/notifications/issues/58

* SC: We need to resolve the issue on discovering channel information in order to make the next release of the Notifications Protocol.
* SC: Just to clarify different cases. Storage level channel information applying to all resources that want to use it, and the other resource specific. I think it'd be good to keep these possibilities open. But I suppose the question is how we specify the requirements.

> * SC: The SP states the all resources must include the storageDescription relation (a description resource of Storage). This is a good shortcut: all resources can rely on this, but there are considerations where specific resources (e.g. containers) have different subscription mechanisms.

Reviewer: ```suggestion
* SC: The SP states the all resources must include the `storageDescription` relation (a description resource of `Storage`). This is a good shortcut: all resources can rely on this, but there are considerations where specific resources (e.g., containers) have different subscription mechanisms.
```

* SC: It seems that we have the parts we need, do we agree on those requirements?

> * AC: describedby seems really useful and also generic in that it allows us to define something that doesn't depend on Solid Protocol. you and i discussed on circular dependencies.. so having that separation has a lot of advantages.

Reviewer: ```suggestion
* AC: `describedby` seems really useful, and also generic in that it allows us to define something that doesn't depend on Solid Protocol. You and I discussed on circular dependencies... so having that separation has a lot of advantages.
```

* SC: So, you're at least plus one on resource specific channel information.
* AC: Yes.
* SC: How about storage level?

> * AC: Useful information. I wonder if fro ma client end, I'd find that really useful. On the other hand, i do want to consider the question of notifications protocol not depending on the solid protocol. if we stated that the property that's defined in the solid protocol. that seems like a useful property to look for that cover the entire storage.

Reviewer: ```suggestion
* AC: Useful information. I wonder if from a client end, I'd find that really useful. On the other hand, I do want to consider the question of notifications protocol not depending on the solid protocol. If we stated that the property that's defined in the solid protocol. that seems like a useful property to look for that cover the entire storage.
```
